charter.html

```diff
@@ -704,16 +690,34 @@ <h2 id=about>About this Charter</h2>
 
 <script>
 document.addEventListener("DOMContentLoaded", function() {
-  document.querySelectorAll("[id]").forEach(function(heading) {
-    let permalink = document.createElement("a");
-    permalink.href = "#" + heading.getAttribute("id");
-    permalink.className = "permalink";
-    permalink.appendChild(document.createTextNode("#"));
-    if (heading.localName == "tr") {
-      heading = heading.children[0];
+  let sectionCounters = [NaN, 0, 0, 0, 0, 0, 0];
+  document.querySelectorAll("[id]").forEach(function(elementWithID) {
+    const permalink = document.createElement("a");
+    permalink.href = "#" + elementWithID.getAttribute("id");
+    if (elementWithID instanceof HTMLHeadingElement) {
+      const headingLevel = parseInt(elementWithID.localName.substring(1), 10);
+      sectionCounters[headingLevel]++;
+      for (let i = headingLevel+1; i < 7; i++) {
```

Comment:
Consider replacing lines 700 through 706 with:

```
      sectionCounters.fill(0, headingLevel + 1);
      let sectionNumber = sectionCounters.slice(0, headingLevel + 1).filter(item=> item != 0);
```

The `slice` call is actually not necessary but may make the intent more clear to the reader.

Reply:
Thanks!